src/components/implementation/no_interface/cFE_booter/osfilesys.h

```diff
@@ -1,113 +1,258 @@
-#include "cFE_util.h"
 #include <string.h>
+#include <dirent.h>
+
 #include "gen/common_types.h"
 #include "gen/osapi.h"
 #include "gen/osapi-os-filesys.h"
 
+#include <cos_component.h>
+#include <cos_defkernel_api.h>
+#include <cos_kernel_api.h>
+
+#include "cFE_util.h"
+
 // Not to be confused with similar OSAL constants.
 // These are only to define size of statically allocated data
-#define MAX_NUM_FS 3
+#define MAX_NUM_FS NUM_TABLE_ENTRIES
 #define MAX_NUM_FILES 100
+#define MAX_NUM_DIRENT 10
 
-#define TAR_BLOCKSIZE 512
-#define INT32_MAX 0x7FFFFFF //2^31 - 1
+// a page is 4096, size of f_part is 5 values * 4 bytes
+#define F_PART_DATA_SIZE (4096-sizeof(struct f_part))
 
-//should be overwritten by linking step in build process
-__attribute__((weak)) char _binary_cFEfs_tar_size=0;
-__attribute__((weak)) char _binary_cFEfs_tar_start=0;
-__attribute__((weak)) char _binary_cFEfs_tar_end=0;
+// currently we compile composite and cFE with different libc
+// muslc for cos and whatever ubuntu32 14 use for cFE
+// these have different structs so I need to implement a dirent struct in cos
+// with the same layout as cFE's dirent
+struct hack_dirent
+{
+    ino_t d_ino;
+    // off_t d_off; these are the 8 bytes that I am removing
+    unsigned short d_reclen;
+    unsigned char d_type;
+    char d_name[256];
+};
 
-//locations and size of tar
-char      *start;
-char      *end;
-size_t    size;
+// for exactly the same reason as above, we need to declare a stat structure to pass
+// back to the cFE that it will recoginize
+struct hack_stat
```

Comment:
Consider putting TODO's everywhere `hack_*` symbols are used — should be a pretty simple `grep`.